class.smtp.php

```diff
@@ -272,28 +272,31 @@ public function connect($host, $port = null, $timeout = 30, $options = array())
         $errstr = '';
         if ($streamok) {
             $socket_context = stream_context_create($options);
-            //Suppress errors; connection failures are handled at a higher level
-            $this->smtp_conn = @stream_socket_client(
+            set_error_handler('SMTP::errorHandler');
```

Comment:
I think it would be better to use `self` here, and `static` in PHPMailer 6, instead of `SMTP` so that it can be overridden. Actually `classname::methodname` was only added as a callable pattern from PHP 5.2.3, and officially PHPMailer 5.2.x still supports 5.0! You can do `array('classname', 'methodname')` instead.